src/Data/List/Membership/Setoid/Properties.agda

```diff
@@ -50,6 +52,28 @@ module _ {c ℓ} (S : Setoid c ℓ) where
   ∉-resp-≋ : ∀ {x} → (x ∉_) Respects _≋_
   ∉-resp-≋ xs≋ys v∉xs v∈ys = v∉xs (∈-resp-≋ (≋-sym xs≋ys) v∈ys)
 
+------------------------------------------------------------------------
+-- Irrelevance
+
+module _ {c ℓ} (S : Setoid c ℓ)
+               (≈-irrelevant : B.Irrelevant (Setoid._≈_ S)) where
+
+  open Setoid S
+  open Unique S
+  open Membership S
+
+  private
+    _≉_ = λ x y → ¬ (x ≈ y)
+
+    ∉×∈⇒≉ : ∀ {x y xs} → All (y ≉_) xs → x ∈ xs → x ≉ y
+    ∉×∈⇒≉ = All.glookup λ y≉z x≈z x≈y → y≉z (trans (sym x≈y) x≈z)
+
+  irrelevant : ∀ {xs} → Unique xs → U.Irrelevant (_∈ xs)
```

Comment:
Do you think we should rename the lemmas `unique⇒irrelevant`? The uniqueness is quite crucial.

Reply:
Also the setoid version should be added to the CHANGELOG.